### DISCUSSION TITLE
Обращение к верхнему уровню в Селект-Дереве

### DISCUSSION THREAD

**Discussion by Makhmut2445** [totumonline/totum-ru-issues-and-discussions#126] [Как это сделать?] [answered]
![image](https://user-images.githubusercontent.com/103171127/165950251-bbd20b87-67e8-4a88-88d2-e6529caebc90.png)
Во вторую строчку мне удалось подтянуть данные родителя, а в остальные подкатегории не получается.. как можно это сделать, чтоб она всегда брала данные от верхнего своего уровня?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
### [Видео на YouTube —>](https://youtu.be/nztH_XulOSQ)

Коды действия в `tree`:
```
a1= : reCalculate(table: $#ntn; where: 'id' = #tree)
a2= : reCalculate(table: $#ntn; where: 'tree_children' = #tree)

ch1= : reCalculate(table: $#ntn; where: 'id' = #old.tree)
ch2= : reCalculate(table: $#ntn; where: 'tree_children' = #old.tree)

dl1= : reCalculate(table: $#ntn; where: 'id' = #old.tree)
dl2= : reCalculate(table: $#ntn; where: 'tree_children' = #old.tree)
```

Код в `tree children`:
```
= : if(condition: #checkbox = true; then: $calc; else: json`[]`)

calc: listAdd(list: $ch; item: #id)

ch: selectTreeChildren(table: $#ntn; id: #id; parent: 'tree')
```

Код в `data`:
```
= : if(condition: #checkbox = false; condition: #level > 0; then: $calc)

calc: select(table: $#ntn; field: 'control_data'; where: 'tree_children' = #tree)
```

#### В видео показан поиск уровня дерева через `#l.tree` в `Расчетных таблицах` в которых `дерево` построено `по текущей таблице` это имеет особенность:

Результатом работы `#l.tree` будет 0 для любого уровня, кроме верхнего — для того, что бы уровень работал корректно, необходимо, что бы колонка с деревом была полностью рассчитана — для этого включите параметр таблицы [Поколоночный пересчет](https://ru.docs.totum.online/tables#table-settings-calculate_by_columns)
